src/pages/CocosGame.tsx

```diff
@@ -60,9 +60,18 @@ const CocosGame: React.FC = () => {
 
   const push = () => {
     const urlParams = new URLSearchParams(window.location.search);
-    history.replace(state.from ?? PAGES.HOME);
-    if (!!urlParams.get("isReload")) {
-      window.location.reload();
+    let fromPath: string = state?.from;
+
+    if (!!fromPath) {
+      if (!!urlParams.get("isReload")) {
+        if (fromPath.includes("?"))
+          history.replace(fromPath + "&isReload=true" ?? PAGES.HOME);
```

Comment:
Use [searchParams](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) to set the URL params